wpengine/wp-graphql-content-blocks
# fix: prevent fatal errors with DOMHelpers methods by improving typesafety and returning early

## What

This PR:
- Updates the methods in the `DOMHelpers` class to return early if no HTML is passed to the method.
- Improves type safety of the `DOMHelpers` methods by inlining their types to the method signatures. To maintain back-compatiblity, the old methods have been deprecated, with `snake_case` alternatives created with the stricter signatures.

> [!IMPORTANT]
> This PR is based on #256 which should be merged first.
> The relevant diff for this PR is: 5b59fe3a68a434748710e8682e1df38c7637bb0f

## Why

Without returning early, numerous DIDOM methods we call internally can fatal error. The improved type safety is to prevent this sort of issue leaking through again in the future.

## To Test

1. Add the following placeholder image block to a page (extracted from a TT4 pattern):

   ```html
   
   <figure class="wp-block-image is-resized"><img alt="" style="aspect-ratio:4/3;object-fit:contain;width:150px"/></figure>
   
   ```

2. Query for `CoreBlockImageAttributes`:

   ```graphql
   {
    nodeByUri(uri: "/sample-page") { # change the URI to your page
      ... on NodeWithEditorBlocks {
        editorBlocks {
          __typename
          ... on CoreImage {
            attributes {
              width
              alt
            }
          }
        }
      }
     }
   }
   ```

3. Ensure no errors, and `attributes` are returned.
  
## Screenshots
-  Before:
![image](https://github.com/user-attachments/assets/be246e3e-1052-4d94-97de-c034aa35cac8)

- After:
![image](https://github.com/user-attachments/assets/4be9f0b1-ffb0-4107-a489-f0278c6ff786)
